Refined the UI and added UI components

Anam-Ashraf7 opened the pull request:
Here are the changes:

- Added progress bar to profile page
- Added a responsive navbar for mobile
- Added filter in home page
- Added settings route for settings section

Anam-Ashraf7 commented:
@CodiumAI-Agent /review

QodoAI-Agent commented:
## PR Review

<table>
<tr>
<tr><td> ⏱️&nbsp;<strong>Estimated&nbsp;effort&nbsp;to&nbsp;review [1-5]</strong></td><td>

4, due to the extensive changes across multiple files and components, which include both UI and functionality updates. The PR touches on various aspects such as CSS, HTML, and React components, requiring a detailed review to ensure consistency and functionality.


</td></tr>
<tr><td> 🧪&nbsp;<strong>Relevant tests</strong></td><td>

No


</td></tr>
<tr><td rowspan=2> 🔍&nbsp;<strong>Possible issues</strong></td>
<td>

<strong>Possible Bug:</strong> The inline style `style="margin-right: 0px !important;"` in `frontend/index.html` might not be necessary or could lead to unexpected layout issues. Inline styles can also make future maintenance harder.</td></tr>
<tr>
<td>

<strong>Consistency Issue:</strong> The PR includes changes to border styles (from `border-2` to `border`) across various components. This needs to be reviewed for visual consistency across the application.</td></tr>
<tr><td> 🔒&nbsp;<strong>Security concerns</strong></td><td>

No

</td></tr>
</table>


<details><summary> <strong>Code feedback:</strong></summary>

<hr><table><tr><td>relevant file</td><td>frontend/src/components/PostProject.jsx
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Consider using CSS classes for styling instead of inline styles for better maintainability and separation of concerns. For example, replace inline styles in JSX elements with className usage. [important]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S46_Anam_Capstone_WeCollab/pull/31/files#diff-bae0270b6bd291dc8f2e4fc4e2e54d63ed5d0995487949797d955ce266a2723cR124'><form className="flex flex-col gap-3 lg:gap-5 bg-white py-10 lg:pb-20 lg:w-[30dvw] md:w-[30dvw] border-black lg:border-r "></a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>frontend/index.html
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Avoid using `!important` in inline styles as it can lead to difficulties in maintenance and override issues in CSS. Consider managing this style through a CSS class. [important]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S46_Anam_Capstone_WeCollab/pull/31/files#diff-959f9cada636604db33bc1ba82fed347457dcb032c37ac195b4343c5f0ea6e72R9'><body style="margin-right: 0px !important;"></a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>frontend/src/components/Form.jsx
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Ensure that form elements have proper accessibility attributes and validation. For instance, adding `aria-label` or `aria-labelledby` can improve accessibility. [medium]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S46_Anam_Capstone_WeCollab/pull/31/files#diff-9b6d85e773821bb24f1c6ace137e9d7ce1b24c5fc04b4e0248dd93e48e3ce568R119'>className={`font-medium py-2 px-2 rounded-md ${formik.touched.name && formik.errors.name ? "border-red-500 border-2" : ""} border-black border`}</a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>frontend/src/components/Home.jsx
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Use React Router's `NavLink` or similar to manage active state instead of manual click handlers for better performance and less code. This can also make the component cleaner and more maintainable. [medium]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S46_Anam_Capstone_WeCollab/pull/31/files#diff-dfb5d804219467626c0f71466cbe81ec8d18cf3ec5d2eeb1ad97596486fa8c7eR18'><SideNavbar/></a></td></tr></table><hr>

</details><hr>

<details> <summary><strong>✨ Review tool usage guide:</strong></summary><hr> 

**Overview:**
The `review` tool scans the PR code changes, and generates a PR review which includes several types of feedbacks, such as possible PR issues, security threats and relevant test in the PR. More feedbacks can be [added](https://pr-agent-docs.codium.ai/tools/review/#general-configurations) by configuring the tool.

The tool can be triggered [automatically](https://pr-agent-docs.codium.ai/usage-guide/automations_and_usage/#github-app-automatic-tools-when-a-new-pr-is-opened) every time a new PR is opened, or can be invoked manually by commenting on any PR.
- When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L23) related to the review tool (`pr_reviewer` section), use the following template:
```
/review --pr_reviewer.some_config1=... --pr_reviewer.some_config2=...
```
- With a [configuration file](https://pr-agent-docs.codium.ai/usage-guide/configuration_options/), use the following template:
```
[pr_reviewer]
some_config1=...
some_config2=...
```
    

See the review [usage page](https://pr-agent-docs.codium.ai/tools/review/) for a comprehensive guide on using this tool.


</details>

Anam-Ashraf7 commented:
Here are the changes:

Added progress bar to profile page
Added a responsive navbar for mobile
Added filter in home page
Added settings route for settings section
